### PR TITLE
[gpgmm/vulkan-headers/vulkan-loader] Fix file conflicts

### LINKED ISSUE #25799: [vulkan-headers] Update to 1.3
Library name:
vulkan
vulkan-headers

New version number:
1.3.221

Other information that may be useful (release notes, etc...)
The current version of vulkan-headers (1.2.203) does not support Vulkan 1.3 applications

### PR DESCRIPTION
**Pull request by JackBoosY**
1. gpgmm uses internal submodules `vulkan-headers` and `vulkan-loader`.
Add new port `vulkan-loader` and rebuild dependencies.
2. Fix gpgmm's binary build directory.
3. Fix gpgmm's header file installation process.
4. Update `vulkan-header` to `1.3.221` to fit the current gpgmm version.
5. Refactor `vulkan-header` build process to use official cmake.

Related: https://github.com/microsoft/vcpkg/runs/7372647110

### CODE REVIEW

File: ports/vulkan-loader/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Only provides dynamic library
+set(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)
```
Comment (JackBoosY) [inline ports/vulkan-loader/portfile.cmake:2]: This port only provides the dynamic library and uses the vulkan generic declaration to use it.
See [official doc](https://github.com/KhronosGroup/Vulkan-Loader/blob/master/docs/LoaderApplicationInterface.md#directly-linking-to-the-loader).

**Comment by JackBoosY** [comment]
@BillyONeal 
Our pipeline tests skip vulkan tests in https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt#L1127-L1135.
Should we add Vulkan SDK installation in our image to avoid the cascade issue?
The required Vulkan SDK version is 1.3.221, link: https://sdk.lunarg.com/sdk/download/1.3.216.0/windows/VulkanSDK-1.3.216.0-Installer.exe

File: ports/vulkan-headers/portfile.cmake [hunk 2]
```diff
@@ -2,14 +2,24 @@
...
     HEAD_REF master
 )
 
-# This must be vulkan as other vulkan packages expect it there.
-file(COPY "${SOURCE_PATH}/include/vulkan/" DESTINATION "${CURRENT_PACKAGES_DIR}/include/vulkan")
-file(COPY "${SOURCE_PATH}/include/vk_video/" DESTINATION "${CURRENT_PACKAGES_DIR}/include/vk_video")
+file(TO_CMAKE_PATH "$ENV{VULKAN_SDK}" VULKAN_DIR)
+get_filename_component(VULKAN_VERSION "${VULKAN_DIR}" NAME)
+if (VULKAN_VERSION VERSION_LESS 1.3.216)
+    message(FATAL_ERROR "${PORT} requires to install Vulkan SDK version greater equal than 1.3.216")
```
Comment (JackBoosY) [inline ports/vulkan-headers/portfile.cmake:13]: We have to check the sdk version because the vulkan-headers match the SDK.

**Review by Cheney-W — DISMISSED** [review]

File: ports/vulkan-loader/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  "version": "1.3.221",
+  "description": "Provides the Khronos official Vulkan ICD desktop loader for Windows, Linux, and MacOS.",
+  "homepage": "https://github.com/KhronosGroup/Vulkan-Loader",
+  "license": "Apache-2.0",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    "vulkan-headers"
```
Comment (vicroms) [inline ports/vulkan-loader/vcpkg.json:12]: Should `vulkan-headers` match this port's version? 

```suggestion
    {
        "name": "vulkan-headers",
        "version>=": "1.3.221"
    }
```

File: ports/vulkan-headers/portfile.cmake (hunk 2 above)
Comment (vicroms) [reply ports/vulkan-headers/portfile.cmake:13]: there's `vulkan` that just checks whether Vulkan SDK is installed, it has a comment in the top stating there are complications installing Vulkan SDK from a port. But I don't know what those are myself.

**Comment by BillyONeal** [comment]
Should we just revert https://github.com/microsoft/vcpkg/pull/25650 for now?

**Comment by JackBoosY** [comment]
This PR will re-add port gpgmm.

File: ports/vulkan-headers/portfile.cmake (hunk 2 above)
Comment (JackBoosY) [reply ports/vulkan-headers/portfile.cmake:13]: > Shouldn't there be a port `vulkan-sdk` which installs this stuff ?

No, port `vulkan` just check the SDK whether it exists.

**Comment by Cheney-W** [comment]
Waiting for VM update.

**Comment by BillyONeal** [comment]
I don't see an actual VM update request here. There are discussions above about installing some sort of SDK into the VMs but I don't know exactly what is being requested; please update scripts\azure-pipelines files with the changes you want.

**Pull request closed without merging**